dev.exs

```diff
@@ -28,6 +28,88 @@ Application.put_env(:phoenix_live_dashboard, DemoWeb.Endpoint,
   ]
 )
 
+defmodule DemoWeb.History do
+  use GenServer
+  alias Phoenix.LiveDashboard.TelemetryListener
+
+  @history_buffer_size 50
+
+  def data(metric) do
+    GenServer.call(__MODULE__, {:data, metric})
+  end
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  def init(metrics) do
+    GenServer.cast(__MODULE__, {:metrics, metrics})
+    {:ok, %{}}
+  end
+
+  def handle_cast({:metrics, metrics}, _state) do
+    {:noreply,
+     for metric <- metrics, reduce: %{} do
+       acc ->
+         key_metrics = Map.get(acc, event(metric.name), [])
+         metric_map = %{metric: metric, history: CircularBuffer.new(@history_buffer_size)}
+         attach_handler(metric, length(key_metrics))
+
+         Map.merge(acc, %{event(metric.name) => [metric_map | key_metrics]})
```

Comment:
You probably want a single map with the metrics as the key and the buffer as value.